notifications/drf_views.py

```diff
@@ -1,29 +1,67 @@
 from datetime import datetime, timedelta, timezone
 from django.db.models import Q
 from django_filters import rest_framework as filters
+from django_filters.widgets import CSVWidget
 from rest_framework.authentication import TokenAuthentication
 from rest_framework.permissions import IsAuthenticated
 from rest_framework import viewsets
 from .models import SurgeAlert, Subscription
+from deployments.models import MolnixTag
 from .serializers import (
     SurgeAlertSerializer,
 #   UnauthenticatedSurgeAlertSerializer,
     SubscriptionSerializer,
 )
 
 
+class CharInFilter(filters.BaseInFilter, filters.CharFilter):
+    pass
+
+
 class SurgeAlertFilter(filters.FilterSet):
     atype = filters.NumberFilter(field_name='atype', lookup_expr='exact')
     category = filters.NumberFilter(field_name='category', lookup_expr='exact')
     event = filters.NumberFilter(field_name='event', lookup_expr='exact')
+    molnix_tags = filters.ModelMultipleChoiceFilter(
+        label='tag-ids',
+        method='molnix_tags_filter',
+        help_text='Molnix_tag GO identifiers, comma separated',
+        widget=CSVWidget,
+        queryset=MolnixTag.objects.all(),
+    )
+    molnix_tag_names = CharInFilter(
+        label='tag-names',
+        method='molnix_tag_names_filter',
+        help_text='Molnix_tag names, comma separated',
+        widget=CSVWidget,
+        # How can it work without queryset??? And if that can work, why that one ^ not?
+    )
+
+    def molnix_tags_filter(self, qs, name, value):
+        if value:
+            return qs.filter(molnix_tags__in=value).order_by('id').distinct()
+        return qs
+
+    def molnix_tag_names_filter(self, qs, name, value):
+        if value:
+            return qs.filter(molnix_tags__name__in=value).order_by('id').distinct()
+        return qs
```

Comment:
I checked and looks like we can also do this in this way.

For the order_by('id'), we shouldn't do ordering in filters as we have separate module for ordering.

We should move `CharInFilter` to a `api/utils.py` or maybe create a new one `main/filters.py`
```suggestion
    molnix_tags = filters.ModelMultipleChoiceFilter(
        label='tag-ids',
        field_name='molnix_tags',
        lookup_expr='in',
        help_text='Molnix_tag GO identifiers, comma separated',
        widget=CSVWidget,
        queryset=MolnixTag.objects.all(),
    )
    molnix_tag_names = CharInFilter(
        label='tag-names',
        field_name='molnix_tags__name',
        lookup_expr='in',
        help_text='Molnix_tag names, comma separated',
        widget=CSVWidget,
    )

    @property
    def qs(self):
        return super().qs.distinct()
```